helpers.go

```diff
@@ -357,6 +357,13 @@ func (s *Spec) GetImagePost(target string) *PostInstall {
 	return nil
 }
 
+func (s *Spec) GetArtifacts(targetKey string) *Artifacts {
```

Comment:
Can we always use a value object?
This way we don't need to deal with a nil everywhere and code can continue to work the way it was before (mostly).

```suggestion
func (s *Spec) GetArtifacts(targetKey string) Artifacts {
```

Reply:
Also looks like `Artifacts.IsEmpty` could use a nil check anyway since a nil artifacts should be considered empty.
This is causing a panic currently in the debian side of things.

Though returning a zero value of `Artifacts` here instead of nil should take care of the panic, I think it makes sense to add the nil check directly to `Artifacts.IsEmpty` as well.